### ISSUE TITLE
Feature request: a flag to tell F* where Z3 is

### ISSUE THREAD

**Issue by cpitclaudel** [FStarLang/FStar#939]
**TL;DR**: it would be great to be able to call `fstar.exe` with a `--z3` flag that tells it where to get `z3`.  Then I would add a corresponding setting to F*-mode (which would default to searching for `z3` in the user's path, or course).  Would there be anything wrong with that ?

**Background:** I'm trying to run F* on a remote machine.  I've updated F* mode to be able to talk to F* over SSH, but I'm running into an issue: my remote machine doesn't have Z3 in its global path, and I don't have a way to add it easily (I'm not root there).

(Of course it's easy enough to update my remote `.profile` to add local directories to my user path… but that doesn't work because OPAM also adds itself to `.profile`, and OPAM is [broken](https://github.com/ocaml/opam/issues/2899).  I can fix the issue locally, but I expect others will run into that too)

**Comment by mtzguido** [comment]
I believe `--smt` will give you that!

**Comment by cpitclaudel** [comment]
!! :)
